kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/SessionViewModel.kt

```diff
@@ -17,92 +15,18 @@ internal class SessionViewModel
     constructor() : ViewModel() {
         @Inject
         internal lateinit var repo: Repository
-        private val _favoriteResourcesLiveData = MutableLiveData<HashSet<String>>(HashSet())
         private val _serviceStatusLiveData = MutableLiveData<State>()
         private val _resourcesLiveData = MutableLiveData<List<Resource>>(emptyList())
-        private var showOnlyFavorites: Boolean = false
 
-        val favoriteResourcesLiveData: MutableLiveData<HashSet<String>>
-            get() = _favoriteResourcesLiveData
         val serviceStatusLiveData: MutableLiveData<State>
             get() = _serviceStatusLiveData
         val resourcesLiveData: MutableLiveData<List<Resource>>
             get() = _resourcesLiveData
 
-        private val favoriteResources: HashSet<String>
-            get() = favoriteResourcesLiveData.value!!
-
         // Actor name
         fun clearActorName() = repo.clearActorName()
 
         fun getActorName() = repo.getActorNameSync()
 
-        fun addFavoriteResource(id: String) {
-            val value = favoriteResources
-            value.add(id)
-            repo.saveFavoritesSync(value)
-            // Update LiveData
-            _favoriteResourcesLiveData.value = value
-        }
```

Comment:
We already have `repo` in here. Instead of moving this code to the view, can we just re-implement `addFavoriteResource` etc by accessing `Repository`?

```suggestion
        fun addFavoriteResource(id: String) {
	    repo.addFavoriteResource(id)
	}
```

Reply:
Ideally, I think the `repo` should be a private member of the view model. Reaching into it kind of defeats its purpose. We don't have any unit-tests right now but that is where we would add them but that is not possible if the view doesn't actually use a defined public API in the view model.

Reply:
Hm I was thinking we could get rid of the view model entirely if possible.
If the view model just wraps methods from the model, couldn't unit tests test the model directly too?

Reply:
> Hm I was thinking we could get rid of the view model entirely if possible. If the view model just wraps methods from the model, couldn't unit tests test the model directly too?

When you say "model", what part are you referring to? I don't think we have a model that represents the entire session, right? We have individual resource models and the repository.

At the moment, the `SessionActivity` still contains a fair bit of "view-logic", like making decision, which resources are actually rendered, whether we show the tabs, etc.

All this logic should IMO be in the view model so that we can / could write a unit-test that pretends to be the UI and checks, whether the view model computes the correct things, for example:

- Init a view model
- Call a function to mark a resource as favourite
- Assert that the tab-layout visibility is `VISIBLE`

Or:

- init a view model without any resources favourited
- Assert that the tab-layout visibility is `GONE`

In other words, the code in `SessionActivity` should be so simple (i.e. no conditions) that it doesn't need testing.

Reply:
By model I meant `Repository`, as in model-view architecture

Reply:
Happy to pair on it if you want :)

Reply:
Oh I see what you mean, there shouldn't be business logic in the view.

Yeah that does justify a view model. I'll flip it back